newIDE/app/src/MainFrame/TabsTitlebar.js

```diff
@@ -27,24 +55,148 @@ const styles = {
     width: 34,
     height: 34,
   },
+  paper: {
+    padding: '8px 10px',
+    minWidth: 180,
+  },
+  tabIcon: {
+    marginLeft: 4,
+    marginRight: 4,
+    display: 'flex',
+  },
+  emptyTabIcon: {
+    marginLeft: 4,
+    marginRight: 4,
+    height: 20,
+    width: 24,
+    display: 'flex',
+  },
+  tooltip: {
+    zIndex: 3,
+    maxWidth: 'min(90%, 300px)',
+  },
+};
+
+type Props = {|
+  anchorElement: HTMLElement,
+  editorTab: EditorTab,
+|};
+
+const Tooltip = ({ anchorElement, editorTab }: Props) => {
+  const gdevelopTheme = React.useContext(GDevelopThemeContext);
+  const [tooltipStyle, setTooltipStyle] = React.useState<Object>(
+    styles.tooltip
+  );
+
+  const brightness = gdevelopTheme.palette.type === 'dark' ? 0.978 : 0.224;
+
+  const editorTabMetadata = getEditorTabMetadata(editorTab);
+
+  React.useEffect(
+    () => {
+      const timeoutId = setTimeout(() => {
+        setTooltipStyle(currentStyle => ({
+          ...currentStyle,
+          transition: 'transform 150ms ease-in-out',
+        }));
+      }, 100);
+      return () => clearTimeout(timeoutId);
+    },
+    // Apply transition after first display of tooltip to avoid having the
+    // transition weirdly applied from the 0;0 coordinates.
+    []
+  );
+
+  const title = editorKindToLabel[editorTabMetadata.editorKind];
+  const subtitle = [
+    'layout',
+    'layout events',
+    'events functions extension',
+  ].includes(editorTabMetadata.editorKind)
+    ? editorTabMetadata.projectItemName
+    : editorTabMetadata.editorKind === 'custom object' &&
+      editorTabMetadata.projectItemName
+    ? editorTabMetadata.projectItemName.split('::')[1]

```

Comment:
just to be safe, we should probably check the length of the array resulting the split to ensure there are at least 2 elements.
Maybe, this should be in a React.useMemo() to make that logic easier to ready with `if`s ?

Reply:
I don't think using useMemo is interesting here but I simplified the logic